IDE/Espressif/ESP-IDF/examples/wolfssl_server/main/CMakeLists.txt

```diff
@@ -7,4 +7,59 @@ set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DWOLFSSL_USER_SETTINGS")
 set(COMPONENT_SRCS "server-tls.c" "wifi_connect.c")
 set(COMPONENT_ADD_INCLUDEDIRS "." "./include")
 
+if( EXISTS "${CMAKE_HOME_DIRECTORY}/components/wolfssl/" AND EXISTS "$ENV{IDF_PATH}/components/wolfssl/" )
+    #
+    # wolfSSL found in both ESP-IDF and local project - needs to be resolved by user
+    #
+    message(STATUS "")
+    message(STATUS "WARNING: Found components/wolfssl in both local project and IDF_PATH")
```

Comment:
Maybe consider making this an error and not just a warning. Bad/unexpected things could happen because of an unreliable location of the component source files.

Reply:
I had considered making it a failure. But I wanted to to be a "it just works" feature. Sure, the user _may_ have an different version installed, but do you think that should really make it a hard fail? This method we are pointing out which is being used, with a warning in the app (noted above).

It might however, be a good idea in a future revision to determine the versions of each and display that information as well.

Reply:
I've reconsidered this and yes: I'll submit a change to make this a hard fail error. The risks of mixing code from each of the (potentially different version) component directories if the user ignores the warning seems ripe for support headaches.